index.d.ts

```diff
@@ -1,184 +1,261 @@
-import { Dispatch, SetStateAction, ReactNode } from 'react';
-
 declare module 'react-table' {
-    export interface Cell<D> {
-        render: (type: string) => any;
-        getCellProps: () => any;
-        column: Column<D>;
-        row: Row<D>;
-        state: any;
-        value: any;
-    }
+  import { ReactNode, ReactText, Dispatch, SetStateAction } from 'react'
+    export type SortingRule<D> = {
```

Comment:
What do you think of using more descriptive generics? Maybe something like `Data` and `Accessor` or `Key` if we add the second?

Reply:
I actually like the single character types in generics, it seems to be the more prevalent typescript pattern, but I'm also not wedded to it.  If you aren't too bothered, then we can leave it as is, if you strongly want to change it, I'll be good with that too.

Reply:
I'm partial to more descriptive generics as they seem to be easier to follow. However, I'm alright with leaving them as single characters for now.